hypershift-operator/controllers/controlplane.go

```diff
@@ -169,38 +201,27 @@ func (r *OpenShiftClusterReconciler) ensureControlPlane(ctx context.Context, clu
 		}
 	}
 
-	// TODO: Apply the manifests without any external dependencies
+	// Use server side apply for manifestss
 	manifests, err := filepath.Glob(filepath.Join(manifestsDir, "*.yaml"))
 	if err != nil {
 		return fmt.Errorf("failed to read manifest dir: %w", err)
 	}
-	manifestsToApply := make(chan string, len(manifests))
-	applyResults := make(chan error, len(manifests))
-	for w := 0; w < 10; w++ {
-		go func() {
-			for file := range manifestsToApply {
-				cmd := exec.Command("/bin/sh", "-c", fmt.Sprintf("oc apply --namespace %s -f %s", name, file))
-				cmd.Env = append(cmd.Env, "KUBECONFIG="+os.Getenv("KUBECONFIG"), "PATH="+os.Getenv("PATH"))
-				output, err := cmd.CombinedOutput()
-				r.Log.Info(string(output))
-				if err != nil {
-					applyResults <- fmt.Errorf("failed to apply manifest %s: %w", file, err)
-				} else {
-					r.Log.Info("applied manifest", "manifest", file)
-					applyResults <- nil
-				}
-			}
-		}()
-	}
-	for _, manifest := range manifests {
-		manifestsToApply <- manifest
-	}
-	close(manifestsToApply)
 	applyErrors := []error{}
-	for i := 0; i < len(manifests); i++ {
-		result := <-applyResults
-		if result != nil {
-			applyErrors = append(applyErrors, result)
+	for _, manifest := range manifests {
```

Comment:
nit: maybe patchErrors? or manifestsErrors?
In near future It probably worth putting this into its own unit tested function